pallets/registrar/src/lib.rs

```diff
@@ -585,6 +540,152 @@ pub mod pallet {
+            parathread_params: Option<ParathreadParamsTy>,
+            relay_proof_block_number: u32,
+            relay_storage_proof: sp_trie::StorageProof,
+            manager_signature: cumulus_primitives_core::relay_chain::Signature,
+            genesis_data: ContainerChainGenesisData<T::MaxLengthTokenSymbol>,
+        ) -> DispatchResult {
+            let account = ensure_signed(origin)?;
+            if !T::AllowRegisterWithRelayProof::get() {
+                return Err(Error::<T>::RegisterWithRelayProofDisabledInConfig.into());
+            }
+            let relay_storage_root = T::RelayStorageRootProvider::convert(relay_proof_block_number)
+                .ok_or_else(|| Error::<T>::RelayStorageRootNotFound)?;
+            let relay_state_proof =
+                GenericStateProof::<cumulus_primitives_core::relay_chain::Block>::new(
+                    relay_storage_root,
+                    relay_storage_proof,
+                )
+                .expect("Invalid relay chain state proof");
+
+            // TODO: move constant to utils:
+            pub const REGISTRAR_PARAS_INDEX: &[u8] = &hex_literal::hex![
+                "3fba98689ebed1138735e0e7a5a790abcd710b30bd2eab0352ddcc26417aa194"
+            ];
+            let bytes = para_id.twox_64_concat();
+            let key = [REGISTRAR_PARAS_INDEX, bytes.as_slice()].concat();
+            // TODO: we don't even need to decode the value, only check if it exists
+            let relay_para_info = relay_state_proof
+                .read_entry::<ParaInfo<
+                    cumulus_primitives_core::relay_chain::AccountId,
+                    cumulus_primitives_core::relay_chain::Balance,
+                >>(key.as_slice(), None)
+                .map_err(|e| match e {
+                    ReadEntryErr::Proof => panic!("Invalid proof provided for registrar paras key"),
+                    e => panic!("Invalid proof provided for registrar paras key: {:?}", e),
+                })
+                .unwrap();
+            let relay_manager = relay_para_info.manager;
+
+            // Verify manager signature
+            // Should include:
+            // * para_id, in case the manager has more than 1 para in the relay
+            // * accountid in tanssi, to ensure that the creator role is assigned to the desired account
+            // * relay_proof_block_number or something time based, to make the signature expiring
+            //   ^ this can make UX very bad, but maybe we can check a conservative range and not the
+            //   exact block number? eg by using (relay_proof_block_number / 128) as the signature msg
+            // * parathread_params, to avoid registering a parathread as parachain and viceversa
+            //   ^ maybe not needed, the relay manager should trust the tanssi creator
+            // * also don't need genesis_data as part of the signature
+            let signature_msg: Vec<u8> = (para_id, &account, relay_proof_block_number).encode();
```

Comment:
We should indeed speak about this yes, as it is important, A few ideas:

- the signature includes something network-specific (so that a signature cannot be re-used across networks). Maybe the relay block hash against which we are veryfing could be a good sign
- By default by signing the block hash this is expiring, as this is one of our stored last N block hashes